server/src/main/java/org/elasticsearch/common/util/concurrent/EsExecutors.java

```diff
@@ -266,6 +267,27 @@ public static String threadName(final String nodeName, final String namePrefix)
         return "elasticsearch" + (nodeName.isEmpty() ? "" : "[") + nodeName + (nodeName.isEmpty() ? "" : "]") + "[" + namePrefix + "]";
     }
 
+    // to be used in assertions only.
+    public static boolean differentExecutors(Thread thread1, Thread thread2) {
+        assert thread1 != thread2 : "only call for different threads";
+        String thread1Name = executorName(thread1);
+        String thread2Name = executorName(thread2);
+        return thread1Name == null || thread2Name == null || thread1Name.equals(thread2Name) == false;
+    }
+
+    // visible for tests
+    static String executorName(Thread thread) {
+        assert Assertions.ENABLED;
```

Comment:
```suggestion
        assert Assertions.ENABLED;
```

This is a no-op: if assertions are disabled, we don't even run this check :)

Reply:
oh, I fixed that in one place already, thanks.